### DISCUSSION TITLE
Custom Route parser

### DISCUSSION THREAD

**Discussion by the-ginger-geek** [Milad-Akarie/auto_route_library#1457] [Q&A] [answered]
We recently migrated to v6 of `auto_route`, after the migration we found that the way we determined an initial (startup) route in the previous version of the package stopped working.

Currently we have a method that determines the startup route based on if it's app or web, that logic is irrelevant, what matters is it returns a `PageRouteInfo` object. We take that object and then pass that object to the below code

```
    final routeParser = MyRouteParser(startupRoute.path, widget.router.matcher);
    final routeInformationProvider = MyRouteInformationProvider();
  
    // my material app widget
    MaterialApp.router(
      routerDelegate: widget.router.delegate(),
      routeInformationParser: routeParser, // <-- custom parser
      routeInformationProvider: routeInformationProvider, // <-- custom route info provider
    )
      
    // the custom parser where we pass in the initial route  
    class MyRouteParser extends DefaultRouteParser {
      final String initialRoute;

      MyRouteParser(this.initialRoute, super.matcher);

      @override
      Future<UrlState> parseRouteInformation(RouteInformation routeInformation) {
        if (routeInformation.location != '/') {
          return super.parseRouteInformation(routeInformation);
        } else {
          return super.parseRouteInformation(RouteInformation(
            location: initialRoute,
            state: routeInformation.state,
          ));
        }
      }
    }
```

This part of the code uses a field that was part of the old `PageRouteInfo` called `path`, this is not available anymore, we started using the `routeName` field on the `PageRouteInfo` object to try and migrate the custom parser as well but I think this is why our parser isn't working anymore.
```
MyRouteParser(startupRoute.path, widget.router.matcher)
```

My question is, where can I get the `path` of the route so that I can get our `RouteParser` working again?

**Comment by the-ginger-geek** [comment, marked as answer]
Found a solution

```
    final route = router.routes.firstWhere((element) => element.name == widget.startupRoute.routeName);
    routeParser = MyRouteParser(route.path, router.matcher);
```

**Reply by Milad-Akarie** [reply]
Hello, you can also do router.match(startupRpute)

Not sure I understand whats going on here but why do you need to use a custom parser?

**Reply by the-ginger-geek** [reply]
We have multiple entry points into the app, there isn't only one initial route and I didn't want to navigate from the initial route to the next page so we use the route parser to redirect to the initial route when the app lands on `/` path. Maybe I'm doing it wrong, couldn't find any info on how to do this in another way. I'm open to suggestions.
